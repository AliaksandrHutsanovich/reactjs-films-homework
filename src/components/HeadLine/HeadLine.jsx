import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Typography } from 'antd';
import clsx from 'clsx';

import '!style-loader!css-loader!sass-loader!antd/dist/antd.css'; //eslint-disable-line

import styles from './HeadLine.scss';

const { Title } = Typography;

const HeadLine = ({
  level,
  text,
  weight,
  fontFamily,
}) => (
  <Title
    level={level}
    className={clsx(styles.header, styles[weight], styles[fontFamily])}
  >
    {text}
  </Title>
);

HeadLine.propTypes = {
  level: PropTypes.number,
  text: PropTypes.string.isRequired,
  weight: PropTypes.oneOf([
    'bold',
    'normal',
  ]),
  fontFamily: PropTypes.oneOf([
    'helveticaNeue',
    'arial',
  ]),
};

HeadLine.defaultProps = {
  level: 1,
  weight: 'normal',
  fontFamily: 'helveticaNeue',
};

export default memo(HeadLine);
