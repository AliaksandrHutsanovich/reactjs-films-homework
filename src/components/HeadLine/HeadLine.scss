@font-face{
  font-family: 'HelveticaNeue';
  src: url('../../assets/fonts/HelveticaNeue.ttf') format('truetype');
}

@font-face {
  font-family: 'HelveticaNeueCondensedBlack';
  src: url('../../assets/fonts/HelveticaNeueCondensedBlack.ttf') format('truetype');
}

.header {
  color: white !important;
}

.bold {
  font-weight: 600 !important;
}

.normal {
  font-weight: 100 !important;
}

.helveticaNeue {
  font-family: HelveticaNeue;
}

.arial {
  font-family: Arial;
}
