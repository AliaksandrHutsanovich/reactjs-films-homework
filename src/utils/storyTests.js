const storyTests = (stories, renderer) => {
  Object.keys(stories).forEach((storyKey) => {
    if (typeof stories[storyKey] !== 'function') {
      return;
    }
    it(`${storyKey} should be rendered properly`, () => {
      renderer.render(stories[storyKey]());
      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
};

export default storyTests;
