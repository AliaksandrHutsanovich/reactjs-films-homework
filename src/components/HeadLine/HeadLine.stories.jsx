import React from 'react';
import HeadLine from './HeadLine';
import image from '../../assets/images/story-background.png';

export default {
  component: HeadLine,
  title: 'components/HeadLine',
  decorators: [
    (story) => (
      <div
        style={{
          background: `url(${image}) no-repeat`,
          padding: '20px',
          width: '800px',
          height: '800px',
        }}
      >
        {story()}
      </div>
    ),
  ],
};

export const headerBoldH1 = () => (
  <HeadLine text="Header h1" weight="bold" />
);

export const headerNormalH2 = () => (
  <HeadLine text="Header h2" level={2} />
);

export const headerNormalH3 = () => (
  <HeadLine text="Header h3" level={3} />
);

export const condensedHeaderNormalH3 = () => (
  <HeadLine text="Header h3" level={3} fontFamily="arial" />
);

export const blueHeaderNormalH3 = () => (
  <HeadLine text="Header h3" level={3} fontFamily="arial" type="blue" align="center" />
);
