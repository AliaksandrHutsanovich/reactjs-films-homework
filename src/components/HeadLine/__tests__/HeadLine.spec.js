import ShallowRenderer from 'react-test-renderer/shallow';
import storyTests from '../../../utils/storyTests';
import * as stories from '../HeadLine.stories';

describe('<HeadLine />', () => {
  const renderer = new ShallowRenderer();

  storyTests(stories, renderer);
});
