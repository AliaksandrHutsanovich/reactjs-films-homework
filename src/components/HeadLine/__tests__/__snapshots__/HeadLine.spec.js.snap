// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeadLine /> condensedHeaderNormalH3 should be rendered properly 1`] = `
<Title
  className="header undefined arial"
  level={3}
>
  Header h3
</Title>
`;

exports[`<HeadLine /> headerBoldH1 should be rendered properly 1`] = `
<Title
  className="header bold undefined"
>
  Header h1
</Title>
`;

exports[`<HeadLine /> headerNormalH2 should be rendered properly 1`] = `
<Title
  className="header undefined undefined"
  level={2}
>
  Header h2
</Title>
`;

exports[`<HeadLine /> headerNormalH3 should be rendered properly 1`] = `
<Title
  className="header undefined undefined"
  level={3}
>
  Header h3
</Title>
`;
